app/scripts/metamask-controller.test.js

```diff
@@ -4155,6 +4180,61 @@ describe('MetaMaskController', () => {
         });
       });
     });
+
+    describe('generateNewMnemonicAndAddToVault', () => {
+      it('generates a new hd keyring instance', async () => {
+        const password = 'what-what-what';
+        jest.spyOn(metamaskController, 'getBalance').mockResolvedValue('0x0');
+
+        await metamaskController.createNewVaultAndRestore(password, TEST_SEED);
+
+        const previousKeyrings =
+          metamaskController.keyringController.state.keyrings;
+
+        await metamaskController.addNewMnemonicToVault(TEST_SEED_ALT);
```

Comment:
Shouldn't this be? (given the test name)
```suggestion
        await metamaskController.generateNewMnemonicAndAddToVault(TEST_SEED_ALT);
```